dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/DeepStorageBrowser.java

```diff
@@ -0,0 +1,117 @@
+
+        final Scene mainScene = new Scene(mainView.getView());
+        primaryStage.getIcons().add(new Image(Main.class.getResource("/images/deep_storage_browser.png").toString()));
+        primaryStage.setScene(mainScene);
+        primaryStage.setMaximized(true);
+        primaryStage.setTitle(resourceBundle.getString("title"));
+        primaryStage.setOnCloseRequest(this::handleWindowClose);
+        primaryStage.show();
+    }
+
+    private static String getPlatformInformation() {
+        return String.format("Java Version: {%s}\n", System.getProperty("java.version"))
+                + String.format("Java Vendor: {%s}\n", System.getProperty("java.vendor"))
+                + String.format("JVM Version: {%s}\n", System.getProperty("java.vm.version"))
+                + String.format("JVM Name: {%s}\n", System.getProperty("java.vm.name"))
+                + String.format("OS: {%s}\n", System.getProperty("os.name"))
+                + String.format("OS Arch: {%s}\n", System.getProperty("os.arch"))
+                + String.format("OS Version: {%s}", System.getProperty("os.version"));
+    }
+
+    private void handleWindowClose(final WindowEvent event) {
+        final ImmutableList<Ds3JobTask> notCachedRunningTasks = jobWorkers.getTasks().stream()
+                .filter(task -> task.getProgress() != 1).collect(GuavaCollectors.immutableList());
+
+        if (Guard.isNullOrEmpty(jobWorkers.getTasks()) || Guard.isNullOrEmpty(notCachedRunningTasks)) {
+            shutdownService.shutdown();
+            event.consume();
+
+        } else {
+
+            final Alert closeConfirmation = createCloseConfirmationAlert(notCachedRunningTasks.size());
+
+            final Optional<ButtonType> closeResponse = closeConfirmation.showAndWait();
+
+            closeResponse.ifPresent(response -> {
+                if (response.equals(ButtonType.OK)) {
+                    shutdownService.shutdown();
+                    event.consume();
+                } else if (response.equals(ButtonType.CANCEL)) {
+                    event.consume();
+                }
+            });
+        }
+    }
+
+    private Alert createCloseConfirmationAlert(final int runningTasks) {
+        final Alert closeConfirmation = new Alert(
+            Alert.AlertType.CONFIRMATION,
+            "Are you sure you want to exit?" //TODO put into a resource bundle
+        );
+
+        final Stage stage = (Stage) closeConfirmation.getDialogPane().getScene().getWindow();
+        stage.getIcons().add(new Image(ImageURLs.DEEPSTORAGEBROWSER));
+
+        final Button exitButton = (Button) closeConfirmation.getDialogPane().lookupButton(
+                ButtonType.OK
+        );
+        final Button cancelButton = (Button) closeConfirmation.getDialogPane().lookupButton(
+                ButtonType.CANCEL
+        );
+
+        exitButton.setText("Exit");
+        cancelButton.setText("Cancel");
+
+        if (jobWorkers.getTasks().size() == 1) {
```

Comment:
This code is only called if there are jobs.  So the size of the tasks should never be zero here.